cli/lsp/diagnostics.rs

```diff
@@ -223,43 +222,41 @@ fn ts_json_to_diagnostics(
   )
 }
 
-pub fn generate_ts_diagnostics(
-  state: &ServerStateSnapshot,
-  runtime: &mut JsRuntime,
+pub async fn generate_ts_diagnostics(
+  snapshot: &ServerStateSnapshot,
```

Comment:
I would still prefer `state` here (or if you must `state_snapshot`)...  RA uses the type interchangeably, and at first I was thinking you were replacing the `JsRuntime` with a v8 snapshot and starting up the isolate each call, but then actually paid attention...